node-hub/dora-gr00t/dora_gr00t/main.py

```diff
@@ -0,0 +1,335 @@
+            
+        def get_action(self, inputs):
+            """TODO: Add docstring."""
+            logger.info(f"Mock get_action called with inputs: {list(inputs.keys())}")
+            return {
+                "action_pred": torch.zeros((1, 16, 22))  # Mock action prediction tensor
+            }
+            
+        def apply_transforms(self, inputs):
+            """TODO: Add docstring."""
+            return inputs
+            
+        def unapply_transforms(self, outputs):
+            """TODO: Add docstring."""
+            return {
+                "action.left_arm": np.zeros((16, 7)),
+                "action.right_arm": np.zeros((16, 7)),
+                "action.left_hand": np.zeros((16, 6)),
+                "action.right_hand": np.zeros((16, 6)),
+                "action.waist": np.zeros((16, 3)),
+            }
+        
+        def _check_state_is_batched(self, obs):
+            """TODO: Add docstring."""
+            for k, v in obs.items():
+                if "state" in k and isinstance(v, np.ndarray) and len(v.shape) < 3:  # (B, Time, Dim)
+                    return False
+            return True
+
+
+# Helper functions
+def unsqueeze_dict_values(data: dict[str, Any]) -> dict[str, Any]:
+    """TODO: Add docstring."""
+    unsqueezed_data = {}
+    for k, v in data.items():
+        if isinstance(v, np.ndarray):
+            unsqueezed_data[k] = np.expand_dims(v, axis=0)
+        elif isinstance(v, torch.Tensor):
+            unsqueezed_data[k] = v.unsqueeze(0)
+        else:
+            unsqueezed_data[k] = v
+    return unsqueezed_data
+
+
+def squeeze_dict_values(data: dict[str, Any]) -> dict[str, Any]:
+    """TODO: Add docstring."""
+    squeezed_data = {}
+    for k, v in data.items():
+        if isinstance(v, np.ndarray):
+            squeezed_data[k] = np.squeeze(v, axis=0)
+        elif isinstance(v, torch.Tensor):
+            squeezed_data[k] = v.squeeze(0)
+        else:
+            squeezed_data[k] = v
+    return squeezed_data
+
+
+class DoraGr00tNode:
+    """TODO: Add docstring."""
+    
+    def __init__(
+        self,
+        node: Node,
+        model_path: str = "nvidia/GR00T-N1-2B",
+        embodiment_tag: str = "gr1",
+        data_config: str = "gr1_arms_waist",
+        denoising_steps: int = 4,
+    ):
+        """TODO: Add docstring."""
+        self.node = node
+        self.model_path = model_path
+        self.embodiment_tag = embodiment_tag
+        self.data_config = data_config
+        self.denoising_steps = denoising_steps
+        self.model = None
+        
+        try:
+            logger.info(f"Initializing GR00T model: {self.model_path}")
+            
+            # Set up data config and transform
+            data_config_obj = DATA_CONFIG_MAP[self.data_config]
+            modality_config = data_config_obj.modality_config() if data_config_obj else {}
+            modality_transform = data_config_obj.transform() if data_config_obj else None
+            
```

Comment:
It seems to me that this try except is very big and mainly encompass the initialization of         self.embodiment_tag and self.denoising_steps.

I don't really understand what is the purpose of this try: